### PR TITLE
google-cloud-cpp: simplify excluded test list

### PR DESCRIPTION
**Pull request by coryan**
## Description of changes

The upstream project labels any test that requires contacting
Google Cloud Platform as an `integration-test`. It is easier to
skip the tests by label than explicitly listing each test.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by turion — APPROVED** [review]

File: pkgs/development/libraries/google-cloud-cpp/default.nix [hunk 1]
```diff
@@ -106,16 +105,18 @@ stdenv.mkDerivation rec {
...
   installCheckPhase = lib.optionalString doInstallCheck ''
     runHook preInstallCheck
 
-    # disable tests that contact the internet
-    ctest --exclude-regex '^(${lib.concatStringsSep "|" excludedTests.whole})'
+    # Disable any integration tests, which need to contact the internet.
+    # Also disable the `storage_benchmark_*` tests. With Protobuf < v23.x they
+    # require -DGOOGLE_CLOUD_CPP_ENABLE_CTYPE_WORKAROUND=ON. With
+    # Protobuf  >= v23.x they require They require setting
+    # -DGOOGLE_CLOUD_CPP_ENABLE_CTYPE_WORKAROUND=OFF
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/google-cloud-cpp/default.nix:118]:
```suggestion
    # Disable any integration tests, which need to contact the internet.
    # Also disable the `storage_benchmark_*` tests. 
    # With Protobuf < 23.x they require -DGOOGLE_CLOUD_CPP_ENABLE_CTYPE_WORKAROUND=ON.
    # With Protobuf >= 23.x they require They require setting -DGOOGLE_CLOUD_CPP_ENABLE_CTYPE_WORKAROUND=OFF
```

Version numbers don't contain prefixes like v in the nixpkgs.
Also writing one sentence per line makes it easier to understand especially now that the before and after case are better comparable

**Comment by coryan** [comment]
@dbolduc FYI

**Comment by moekhalil** [comment]
Just floating this idea, any chance this PR can include updating the SDK as well? Looking around here it seems that using the 'update.sh' file will help in updating this PR for updating the package. Package is 452.0.1 and current version is 466.0.0. Not sure of a way to update locally w/o using the official gcloud cli tool.

File: pkgs/development/libraries/google-cloud-cpp/default.nix (hunk 1 above)
Comment (coryan) [reply pkgs/development/libraries/google-cloud-cpp/default.nix]: Done. Please take another look.

**Comment by coryan** [comment]
Thanks for the quick reviews and getting this merged.  I hope it will make it easier to upgrade this package from time to time.

**Pull request merged**